assets/src/components/withDefaultLabel/withDefaultLabel.js

```diff
@@ -0,0 +1,33 @@
+import { Component } from '@wordpress/element';
+import { cloneDeep } from 'lodash';
+
+export function withDefaultLabel( WrappedComponent ) {
```

Comment:
Hmmm, i made it a HOC on purpose so it was composable, It's not tied to `fromThemeOptions` 🤔 You can use this HOC with a `<SelectControl>` component as long as it has `options` and a `defaultValue`.

Reply:
I was thinking of they're fully coupled because the `defaultValue` prop is being set by  `fromThemeOptions`, but you're right that it could be set by anything else or manually.

Still we could keep `withDefaultLabel` HOC and just apply it to `fromThemeOptions` before exporting.

So at the end of `fromThemeOptions`:
```
return compose(
    withDefaultLabel,
)(
    WrappingComponent
)
```

What do you think?

Reply:
Maybe I'm wrong, but I tried this and `withDefaultLabel` is only getting `metaKey`, `label` and `theme`, not `options`, which I think it makes sense since it changes the composing order. I decoupled that on purpose because `withDefaultLabel` is a bit tied to something having `options` with a `label` property, which is not the case for the `ColorPalette` component. Not sure, I don't think we should add a `(default)` label as part of the behavior of `fromThemeOptions`, I see a benefit in making an explicit call to `withDefaultLabel` when it makes sense to add a label.

Reply:
Yeah it would need to be applied after `fromThemeOptions` logic. The problem with all these HOCs is that they do clutter up the code as we have to do all the enhancements in each usage. But I guess we can solve that problem later, so I'll approve this PR now.